frontend/public/components/utils/file-input.tsx

```diff
@@ -28,13 +37,20 @@ export class FileInput extends React.Component<FileInputProps, FileInputState> {
       });
       return;
     }
+    let fileIsBinary = false;
     const reader = new FileReader();
     reader.onload = () => {
-      const input = reader.result;
-      this.props.onChange({
-        fileData: input,
-        fileName: file.name,
-      });
+      const input = reader.result as string;
+      if (containsNonPrintableCharacters(input) && !fileIsBinary) {
```

Comment:
Why is the `!fileIsBinary` check needed?

Reply:
@spadgett we start out assuming we can readAsText, but when we come here after readingAsBinary, we no longer need to drop down into this code or we'll loop.